### PR TITLE
fix(ci): handle release builds in GHA CI workflow

### PR DESCRIPTION
**Pull request by BradLugo**
## Description

Our `tag` make target relies on the `RELEASE_TAG` environment variable being set to the tag name when we perform a release build. Many of our scripts and processes rely on `make tag` outputting the current string (like the release version). We used to set this environment variable properly in OSCI due to:
https://github.com/stackrox/scanner/blob/372710a6a5dcbd786af68572d2c97b76262cd7f8/scripts/ci/lib.sh#L622
https://github.com/stackrox/scanner/blob/372710a6a5dcbd786af68572d2c97b76262cd7f8/scripts/ci/lib.sh#L696-L706
For these changes, I copied the `stackrox/stackrox` pattern for handling tagged builds in our GitHub Action workflow, which was added in this PR: https://github.com/stackrox/stackrox/pull/4239

Note: This needs to be cherry-picked into the `release-2.34` branch.

### CODE REVIEW

**Review by dcaravel — APPROVED** [review]
LGTM, once merged will cherry-pick and re-tag

**Review by RTann — APPROVED** [review]

**Pull request merged**
